internal/typeinfo/arginfo_test.go

```diff
@@ -241,5 +245,27 @@ func (s *typeInfoSuite) TestGenerateArgInfoMapError(c *C) {
 
 	_, _, err = argInfo.AllStructOutputs("myMap")
 	c.Assert(err, Not(IsNil))
-	c.Assert(err.Error(), Equals, "columns must be specified for non-struct type")
+	c.Assert(err.Error(), Equals, `cannot use map with asterisk unless columns are specified`)
+
+	_, err = argInfo.InputSlice("myMap")
+	c.Assert(err, Not(IsNil))
+	c.Assert(err.Error(), Equals, `cannot use slice syntax with map`)
+}
+
+func (s *typeInfoSuite) TestGenerateArgInfoSliceError(c *C) {
```

Comment:
The tests are good but I am wondering if we could instead use table tests which will make it easier to add them later. We would have a test for `AllStructOutput` another for `InputMember`, etc. so sort of the opposite organization. Another benefit is that you can see all the test cases for a function at once making it easier to spot if we are missing any combination.

Reply:
Changed to table driven tests here. I agree that its better.